### ISSUE TITLE
feat: OnlineEvaluationPolicy — test-before-train windowed evaluation

### ISSUE THREAD

**Issue by LakshyAAAgrawal** [gepa-ai/gepa#270]
## Motivation

GEPA's current evaluation flow is **offline**: optimize on a training set, evaluate on a held-out validation set. This is appropriate when you have a fixed labeled dataset.

However, many real-world scenarios are **online**: data arrives in a stream, and you want to optimize the prompt on the fly as new examples come in. The correct evaluation protocol for online optimization is **test-before-train** — measure performance on each example *before* updating the prompt with that example's feedback. This avoids look-ahead bias and reflects true deployment performance.

## Proposal

Add `OnlineEvaluationPolicy` as a built-in policy:

```python
from gepa.strategies import OnlineEvaluationPolicy

result = gepa.optimize(
    ...
    val_evaluation_policy=OnlineEvaluationPolicy(
        window_size=20,         # evaluate on this many examples before each update
        eval_before_train=True, # test-then-train within each window
    ),
)
```

In this mode:
- The dataset is split into consecutive windows
- For each window: first **evaluate** the current candidate on the window (this is the true test), then **train** on those examples to update the candidate
- Cumulative accuracy across all windows is the reported metric

## Why This Is Useful

- Correct evaluation methodology for streaming / production data scenarios
- Enables continuous optimization: as new labeled examples arrive, the prompt adapts
- The "test before train" protocol gives an unbiased estimate of deployment performance
- Useful for monitoring: you can see whether the prompt is improving over time on live data

## Backward Compatibility

Implements the existing `EvaluationPolicy` protocol. Fully opt-in, no changes to existing behavior.
